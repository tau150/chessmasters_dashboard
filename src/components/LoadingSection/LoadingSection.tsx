import { VStack, Spinner, Heading } from "@chakra-ui/react";

export const LoadingSection = () => {
  return (
    <VStack data-testid="loading-section" justify="center">
      <Spinner color="orange.300" size="xl" />
      <Heading color="orange.300" mt="4" size="lg">
        Loading...
      </Heading>
    </VStack>
  );
};
