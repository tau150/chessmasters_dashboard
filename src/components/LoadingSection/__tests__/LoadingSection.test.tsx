import { expect } from "vitest";
import { render } from "@testing-library/react";

import { LoadingSection } from "../LoadingSection";

describe("LoadingSection component", () => {
  it("should render properly", () => {
    const { container } = render(<LoadingSection />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
