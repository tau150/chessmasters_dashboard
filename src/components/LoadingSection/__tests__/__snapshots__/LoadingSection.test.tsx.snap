// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingSection component > should render properly 1`] = `
<div
  class="chakra-stack css-zliwli"
  data-testid="loading-section"
>
  <div
    class="chakra-spinner css-11kyrpg"
  >
    <span
      class="css-8b45rq"
    >
      Loading...
    </span>
  </div>
  <h2
    class="chakra-heading css-7akj9a"
  >
    Loading...
  </h2>
</div>
`;
